packages/protocol/engine/tangle/booker/booker.go

```diff
@@ -108,6 +106,18 @@ func (b *Booker) BlockBookingDetails(block *Block) (pastMarkersConflictIDs, bloc
 	b.evictionMutex.RLock()
 	defer b.evictionMutex.RUnlock()
 
+	block.StructureDetails().PastMarkers().ForEachSorted(func(sequenceID markers.SequenceID, _ markers.Index) bool {
```

Comment:
Do you want to maybe refactor this into a function of the `StructureDetails`? Something like `RLockPartMarkers` and `RUnlockPastMarkers`